src/crm/getTestStation.ts

```diff
@@ -0,0 +1,35 @@
+import pRetry from 'p-retry';
+import dateFormat from 'dateformat';
+import logger from '../observability/logger';
+import { getTestStationEntities } from './dynamicsWebApi';
+import { DynamoTestStation } from './DynamoTestStation';
+import config from '../../config';
+
+export const getTestStations = async (date: Date): Promise<DynamoTestStation[]> => {
+  const ceUrl = config.crm.ceAccountUrl;
+  const modifiedOnDate: string = dateFormat(date, 'yyyy-mm-dd');
+
+  logger.info(`Trying to get test stations informations modified on: ${modifiedOnDate}`);
+
+  const filteredUrl = `${ceUrl}/?$select=accountid,address1_composite,name,emailaddress1,dvsa_premisecodes,dvsa_testfacilitytype,dvsa_accountstatus,address1_latitude,address1_longitude,telephone1,dvsa_openingtimes,modifiedon&$filter=modifiedon%20eq%20${modifiedOnDate}`;
```

Comment:
As we're allowing invocation with a date in the past (not just the default of yesterday, we might need to make the OData query `>=` rather than just `=` - otherwise if we have a week's worth of changes we'd have to invoke it for every day of that missing week..

```suggestion
  const filteredUrl = `${ceUrl}/?$select=accountid,address1_composite,name,emailaddress1,dvsa_premisecodes,dvsa_testfacilitytype,dvsa_accountstatus,address1_latitude,address1_longitude,telephone1,dvsa_openingtimes,modifiedon&$filter=modifiedon%20ge%20${modifiedOnDate}`;
```

Reply:
I'd just be worried say if someone accidentally put in a date from 2019 and it pulled all the records through from dynamics' table to update in dynamodb which could mean sending 100s of events to eventbridge from one data pull from dynamics

Reply:
That's a good point. We probably do want this to be >= because if we want to run from a particular date in the past it's probably because for some reason the service hasn't been running and we want to run it to catch up with any changes that have happened.
We might even want to do this first time we run to make sure all the data in dynamo is updated from dynamics.